source/fe/mapping_q_generic.cc

```diff
@@ -106,7 +106,7 @@ namespace internal
         const double discriminant = b*b - 4*a*c;
         // exit if the point is not in the cell (this is the only case where the
         // discriminant is negative)
-        if (discriminant < 0.0)
+        if (discriminant < 0.0 || ( a == 0.0 && b == 0.0))
```

Comment:
the changes in this file would also have been appropriate for a separate PR. Out of curiosity, would it not have been enough to test for `discriminant <= 0.0`? Or can you think of *legitimate* cases where `discriminant==0` but `!(a==0 && b == 0)`?